### ISSUE TITLE
"flags" is meaningless

### ISSUE THREAD

**Issue by dtolnay** [rust-lang-nursery/bitflags#67]
This threw me off when reading the bitflags documentation for the first time.

```rust
bitflags! {
    flags Flags: u32 {
        /* ... */
    }
}
```

I know it is not a keyword because my IDE does not highlight it like one. The casing suggests that it is a variable name or module name. Nope, it is a nugatory token required by the macro.

I think the important relevant thing to communicate to readers is that a type with the given name is being declared by the macro. Either of the following would be far more valuable I think.

```rust
bitflags! {
    type Flags: u32 {
        /* ... */
    }
}
```

```rust
bitflags! {
    struct Flags: u32 {
        /* ... */
    }
}
```

**Comment by brson** [comment]
Let's change this to "struct". We don't need to preserve backwards compatibility.

**Comment by vedantrathore** [comment]
@brson so I have to change `flags` to `struct` [here](https://github.com/rust-lang-nursery/bitflags/blob/master/src/lib.rs#L114) and [here](https://github.com/rust-lang-nursery/bitflags/blob/master/src/lib.rs#L68) right?

**Comment by brson** [comment]
@vedantrathore yes, but those locations are only in documentation. The most important place to change is in the macro defenition itself, like [here](https://github.com/rust-lang-nursery/bitflags/blob/master/src/lib.rs#L179) and [here](https://github.com/rust-lang-nursery/bitflags/blob/master/src/lib.rs#L196) and [here](https://github.com/rust-lang-nursery/bitflags/blob/master/src/lib.rs#L208).

Once you've changed the macro definition, run `cargo test`. That will test the macro and the documentation examples. If the tests pass you are likely done and can submit a PR. Sorry for taking so long to get back to you. Feel free to ping me on IRC on send an email to get ahold of me faster.

**Comment by vedantrathore** [comment]
@brson no problem :) thanks for the info, I'll get onto it.

**Comment by crumblingstatue** [comment]
I believe this issue can be closed now, see #84.
